test/int/featureBasedSuits/hitBreakpointWithUnusualSourceMap.test.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as _ from 'lodash';
+import { testUsing } from '../fixtures/testUsing';
+import { LaunchProject } from '../fixtures/launchProject';
+import { TestProjectSpec } from '../framework/frameworkTestSupport';
+import { utils } from 'vscode-chrome-debug-core';
+
+suite('Unusual source-maps', () => {
+    testUsing(`file:/// url in sources' field`,
+        async context => {
+            const testSpec = TestProjectSpec.fromTestPath('featuresTests/unusualSourceMaps/fileUrlInSources');
+
+            // Update source-map to have a file:/// url in the sources field
+            const sourceMapPath = testSpec.src('../app.js.map');
+            const sourceMapContents = await utils.readFileP(sourceMapPath);
+            const sourceMapJSON = JSON.parse(sourceMapContents);
+            sourceMapJSON['sources'] = [`file:///${testSpec.src('../app.ts').replace(/\\/g, '/')}`];
```

Comment:
Why do we need to modify the source map here? It looks like `app.js.map` already uses a file:/// url

Reply:
We don't know what the absolute path of app.ts is going to be in the CI machine, so we can't hardcode it. We need to update it dynamically.

Reply:
gotcha. That makes sense